bridge/engine/ProcessUnackedRecordingEventPacketsJob.cpp

```diff
@@ -26,9 +26,15 @@ ProcessUnackedRecordingEventPacketsJob::ProcessUnackedRecordingEventPacketsJob(
 
 void ProcessUnackedRecordingEventPacketsJob::run()
 {
+    uint64_t timestamp = utils::Time::getAbsoluteTime();
+    if (!_unackedPacketsTracker.shouldProcess(timestamp / utils::Time::ms))
```

Comment:
This solves data race with Engine thread

Reply:
It's a bit unfortunate that we have to create jobs that have nothing to do. This is packets have a very low frequency. In some recording we can have long periods (like minutes) without a single packet. It would be nice if we had a fast check in engine thread to not run this if there is zero packets to track

Reply:
I agree. For the video rtx we want to request packets asap, so we have them before the client requests them also.
I guess the only way to signal this is to have an atomic flag set by transport when there is a potential loss detected / need to request. Then that flag will be used to trigger the job. On the other hand, atomic flags have a cost too.